### DISCUSSION TITLE
Javascript Transformer - External DB Datetime Field

### DISCUSSION THREAD

**Discussion by chadgriff** [Budibase/budibase#3659] [Help] [answered]
I am using an external database (MS SQL) where I am storing event information (i.e. eventName, eventStart, eventEnd, etc.).  The start and end fields are set to datetime in the database. The data used to populate the field is coming from a Budibase Date picker. 

I want to display a list of events filtered to just events scheduled to start in the future. I would like to display the eventStart and eventEnd in some other format than ISO 8601 (i.e. 2021-12-02T20:00:00.000Z). I think the way to achieve this is to leverage the new Javascript Transformer functionality built into the MS SQL external data connector. However, my javascript skills are lacking to say the least.  

I would appreciate any examples of how to format the datetime fields using the JS Transformer functionality anyone would be willing to share to get me started. Additionally, I would appreciate any guidance for how to build the proper filter as it appears the ability to create a binding with the "More Than" or "Less Than" operator is not possible when the data is read from the database (assuming that is because the datetime fields are returned as text).

My query is simply as follows :

`Select 
id,
eventName,
eventLocation,
eventNotes,
eventStart,
eventEnd,
allDay,
participants,
emailAttempts,
createdBy,
createdDate,
lastModifiedBy,
lastModifiedDate
From demoData`

Currently, the JSON returned looks like this:
![image](https://user-images.githubusercontent.com/15015476/144531103-6b75a14a-b910-4bb3-b381-a6d2c0a964da.png)

The eventStart and eventEnd data is displaying within the screen as:

![image](https://user-images.githubusercontent.com/15015476/144531725-e60b63e7-1488-4c1c-80b9-f916071d92cd.png)

**Comment by mjashanks** [comment, marked as answer]
Hi Chad.

We have just updated our MS SQL Server connector. You should now be able to "Fetch Tables from Database"...

![image](https://user-images.githubusercontent.com/3524181/144596620-bf8add32-d259-4200-9298-49f4006c80f0.png)

When you do, all your tables will be available for binding on the frontend. 

This will automatically recognise your `Date` column as  dates, and display them in your own locale:

![image](https://user-images.githubusercontent.com/3524181/144596830-d2c7e2b5-4a24-4688-837f-695cebad8ade.png)

You should also be able to properly use `More Than` and `Less Than` to filter dates, using the `Define Filters` button, on your data provider.

I think that this will be the best way to achieve your needs, rather than writing a custom query.

If you need to use a custom query, you can change the type of the field via the `Schema` tab:

![image](https://user-images.githubusercontent.com/3524181/144597248-488f1227-33b6-4c0c-8092-d1c00d7a4ef0.png)

**Reply by chadgriff** [reply]
Thanks Michael. Got the update installed and off and running once again!
